examples/Rust library overview.ipynb

```diff
@@ -0,0 +1,500 @@
+    "\n",
+    "**Paulistring** is a sequence of Pauli letters representing the tensor product of the respective Pauli matrices.\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "39e8ad17",
+   "metadata": {},
+   "source": [
+    "## Basic data structures"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "8450a288",
+   "metadata": {
+    "vscode": {
+     "languageId": "rust"
+    }
+   },
+   "outputs": [],
+   "source": [
+    ":dep syn = { path = \"../.\" } // If you run the syn package locally"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "38e416d0",
+   "metadata": {
+    "vscode": {
+     "languageId": "rust"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "use syn::data_structures::{PauliString, PauliPolynomial, CliffordTableau}\n",
+    "use syn::ir::pauli_exponential::PauliExponential"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "id": "ccbfc999-681a-4083-96a8-9b8b35fa4ade",
+   "metadata": {
+    "vscode": {
+     "languageId": "rust"
+    }
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "I X Y Z\n"
+     ]
+    }
+   ],
+   "source": [
+    "let pauli_string = \"IXYZ\";\n",
+    "let paulivec = PauliString::from_text(pauli_string);\n",
+    "\n",
+    "println!(\"{}\", paulivec);"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "id": "4987d78f",
+   "metadata": {
+    "vscode": {
+     "languageId": "rust"
+    }
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Pauli Polynomial size:4 qubits x 3 terms\n",
+      " | 0.3 0.7 0.12  | \n",
+      "QB0 | I X Y |\n",
+      "QB1 | X X Y |\n",
+      "QB2 | Y I I |\n",
+      "QB3 | Z I I |\n"
+     ]
+    },
+    {
+     "data": {
+      "text/plain": [
+       "()"
+      ]
+     },
+     "execution_count": 4,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "let ham = vec![(\"IXYZ\", 0.3), (\"XXII\", 0.7), (\"YYII\", 0.12)];\n",
+    "let pp = PauliPolynomial::from_hamiltonian(ham);\n",
+    "//println!(\"{}\", pp); // WIP\n",
+    "\n",
+    "println!(\"Pauli Polynomial size:{} qubits x {} terms\", pp.size(), pp.length());\n",
+    "\n",
+    "print!(\" | \");\n",
+    "for i in 0..pp.length(){\n",
+    "    print!(\"{} \", pp.angle(i));\n",
+    "}\n",
+    "println!(\" | \");\n",
+    "for i in 0..pp.size(){\n",
+    "    println!(\"QB{} | {} |\", i, pp.chains()[i]);\n",
+    "}"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "id": "2b395afb",
+   "metadata": {
+    "vscode": {
+     "languageId": "rust"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "use syn::data_structures::PropagateClifford; // Would be nice if this was not needed when importing CliffordTableau"
```

Comment:
Traits always have to be imported. Only alternative would be to re-export common types and traits as a "prelude" module such that users can do `use syn::prelude::*` and get everything. We can discuss this at some point.